flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisProducer.java

```diff
@@ -423,4 +432,43 @@ private void flushSync() throws Exception {
 			}
 		}
 	}
+
+	/**
+	 * Remove references created by the producer, preventing the classloader to unload. References were
+	 * analyzed as of version 0.14.0.
```

Comment:
It would be good to add a version reference for which `aws-java-sdk-core` was used.